src/components/Search.vue

```diff
@@ -0,0 +1,196 @@
+
+<script>
+import { countlock } from '../utils/mixins';
+
+export default {
+  mixins: [countlock],
+  props: {
+    maxQuickResults: {
+      type: Number,
+      default: 6,
+    },
+    showMore: {
+      type: Boolean,
+      default: true,
+    },
+  },
+  inject: ['girderRest'],
+  data() {
+    return {
+      searchText: '',
+      searchMode: 'prefix',
+      searchTypes: ['user', 'folder', 'item'],
+      searchOptionsMenu: false,
+    };
+  },
+  computed: {
+    loading() {
+      return this.locked;
+    },
+    quickResults() {
+      return this.searchResults.slice(0, this.maxQuickResults);
+    },
+    searchParams() {
+      return {
+        q: this.searchText,
+        mode: this.searchMode,
+        types: JSON.stringify(this.searchTypes),
+        // + 1 to determine if total results > maxQuickResults
+        limit: this.maxQuickResults + 1,
+      };
+    },
+  },
+  asyncComputed: {
+    searchResults: {
+      default: [],
+      async get() {
+        let results = [];
+        this.lock();
+        try {
+          if (this.searchText) {
+            const { data } = await this.girderRest.get('resource/search', {
+              params: this.searchParams,
+            });
+            results = [].concat(...this.searchTypes.map(t => data[t]));
```

Comment:
We aren't guaranteed to receive these responses in the order that we sent the requests, i.e. our results might not be for the value currently typed into the search box. Should we do anything to try to resolve that? (Feel free to file an issue, we don't necessarily need to resolve this on this PR.)

Reply:
Yeah, that's true...    

Let's log an issue about it and see how it relates to #83 ?

Reply:
Sounds good to me. I was thinking we could just keep track of the last Promise that was issued and throw out any that we receive that aren't from that one. Will open the issue.

Reply:
I don't think Vuex has utilities to handle this kind of issue. Keep a queue like @zachmullen would work good enough for just this case, but this kind issue generally is asking for things like https://github.com/vuejs/vue-rx

Reply:
#85